frontend/src/app/shared/components/tree/auxiliary/TreeDataObject.ts

```diff
@@ -0,0 +1,5 @@
+export class TreeDataObject {
```

Comment:
Please, create all objects as interfaces


export interface TreeDataObject

classes in JS/TS is mainly for logic, for data they are using interfaces

Reply:
The same for next files